### ISSUE TITLE
Explosions can destroy internal components and attachments of PKAs, possibly other guns.

### ISSUE THREAD

**Issue by Boggart** [tgstation/tgstation#47021]
## Round ID: 
Terry, multiple rounds, IDs unknown.

## Reproduction:

Get a PKA, attach a flashlight and combat knife to it, add some mods. Go to mining, get hit with a gibtonite blast or two. If you're lucky you've lost only the attachments to your pka, if not the cell inside was also deleted and your pka is now useless. The PKAs also had no stated mod capacity, they just displayed the percent symbol and adding a new mod didn't change the total left.

If you had a plasma miner or advanced plasma miner in your other hand it was probably vaporized instantly in the blast. (Possibly related, I've never had one survive an explosion that took out the PKA)

On one round it happened, an admin confirmed that the cell inside both my PKAs were gone.

**Comment by nemvar** [comment]
I blame floydmats but really its more that gun attachment handling isn't really sane in general.

**Comment by necromanceranne** [comment]
This is definitely happening with other weapons. Cells inside guns appear to be vulnerable to explosions, and in the past, so have firing pins.

Potentially the reason why plasma cutters are dying is because quite possibly because they're made from plasma and are still inheriting something from that. 

@Qustinnus looks like something happened. I did actually notice this back when I was testing out the PR initially, so in my original issue you could probably source out the cause when my M1911-P had it's cell deleted from the explosion, which I'd never seen before now.

**Comment by necromanceranne** [comment]
Actually I just tested this. It's every cell. Every cell in the game. This is super exploitable and probably needs a fix and soon, as even small explosions like welderbombs are wiping out energy gun cells. Sec literally has their guns bricked by small explosions. Good lord!

**Comment by Qustinnus** [comment]
cannot reproduce in any way, cells dont have plasma material and neither does anything else in the gun. Could this be an out-of-date server?

**Comment by Jalleo** [comment]
Its more likely someone changed a piece of logic and this has affected the internals for guns. It happened before and got fixed.

**Comment by nemvar** [comment]
Is this still happening?

**Comment by Gousaid67** [comment]
can confirm this is still happening with the cell, i cant manage to destroy the attachements though

**Comment by Mooshimi** [comment]
yeah the cell in the cap's antique can still be destroyed from my personal experience
